data_safe_haven/infrastructure/programs/sre/networking.py

```diff
@@ -1356,10 +1360,13 @@ def __init__(
         )
 
         # Define the virtual network and its subnets
+        # Note that these names for AzureFirewall subnets are required by Azure
         subnet_application_gateway_name = "ApplicationGatewaySubnet"
         subnet_apt_proxy_server_name = "AptProxyServerSubnet"
```

Comment:
Are these used anywhere else?

Reply:
We use them to retrieve the subnets at the end of this function. There are two ways to create subnets in Pulumi: inside the VNet definition or separate from the VNet definition. Creating them at the same time as the VNet means that Pulumi will create/delete them in the right order, but also means that we don't have a handle on the resource. We solve this by running `get_subnet` which needs the subnet name.